### PR TITLE
ssl: fix test code compilation warnings

### PR DESCRIPTION
**Pull request by u3s**

### CODE REVIEW

**Comment by dgud** [comment]
Does this fix all warnings? If so why not turn on warnings as errors in the makefile so we don't introduce new ones.

**Comment by u3s** [comment]
> Does this fix all warnings?
I believe it does.

> If so why not turn on warnings as errors in the makefile so we don't introduce new ones.
I thought about it but didn't dare doing it ;-) @IngelaAndin what do you say about that proposal?

**Comment by IngelaAndin** [comment]
I do not mind having no warnings if possible, I am all for that. I do know that there exist situations in testing where you want to deliberate do things that will cause warnings in able to provoke some kind of error for negative testing. I have not evaluated if we have such code in ssl tests (maybe not). I think it is unreasonable to require all  test code to be waring free, and if you can solve it by suppressing particular warnings that is good too. I think the argument was that some warnings could not be suppressed. But if you got rid off all warnings why not require it for ssl test for now and if we feel we need to have a warning we can cross that bridge when we get there.

**Comment by dgud** [comment]
You can always temporary remove that during development.

**Comment by IngelaAndin** [comment]
> You can always temporary remove that during development.

Sure, this I do not care about, I do not mind fixing the warnings. The problem as I see it is if you want to do a bad-match or something to force something to crash and see how it is handled, at least that  is the argument that is usually made. Maybe there could be some other trick or the particular warning can be suppressed, fine by me.

File: lib/ssl/test/ssl_renegotiate_SUITE.erl [hunk 1]
```diff
@@ -323,7 +323,7 @@ client_no_wrap_sequence_number() ->
     [{doc,"Test that erlang client will renegotiate session when",
      "max sequence number celing is about to be reached. Although"
      "in the testcase we use the test option renegotiate_at"
-     " to lower treashold substantially."}].
+     " to lower treshold substantially."}].
```
Comment (Maria-12648430) [inline lib/ssl/test/ssl_renegotiate_SUITE.erl:326]:
```suggestion
     " to lower threshold substantially."}].
```

File: lib/ssl/test/ssl_renegotiate_SUITE.erl [hunk 2]
```diff
@@ -347,7 +347,7 @@ client_no_wrap_sequence_number(Config) when is_list(Config) ->
 					{host, Hostname},
 					{from, self()},
 					{mfa, {ssl_test_lib,
-					       trigger_renegotiate, [[ErlData, treashold(N, Version)]]}},
+					       trigger_renegotiate, [[ErlData, treshold(N, Version)]]}},
```
Comment (Maria-12648430) [inline lib/ssl/test/ssl_renegotiate_SUITE.erl:350]:
```suggestion
					       trigger_renegotiate, [[ErlData, threshold(N, Version)]]}},
```

File: lib/ssl/test/ssl_renegotiate_SUITE.erl [hunk 3]
```diff
@@ -360,7 +360,7 @@ server_no_wrap_sequence_number() ->
     [{doc, "Test that erlang server will renegotiate session when",
      "max sequence number celing is about to be reached. Although"
      "in the testcase we use the test option renegotiate_at"
-     " to lower treashold substantially."}].
+     " to lower treshold substantially."}].
```
Comment (Maria-12648430) [inline lib/ssl/test/ssl_renegotiate_SUITE.erl:363]:
```suggestion
     " to lower threshold substantially."}].
```

File: lib/ssl/test/ssl_renegotiate_SUITE.erl [hunk 4]
```diff
@@ -536,9 +536,9 @@ renegotiate_rejected(Socket) ->
     ok.
 
 %% First two clauses handles 1/n-1 splitting countermeasure Rizzo/Duong-Beast
-treashold(N, ?TLS_1_0) ->
+treshold(N, ?TLS_1_0) ->
```
Comment (Maria-12648430) [inline lib/ssl/test/ssl_renegotiate_SUITE.erl:539]:
```suggestion
threshold(N, ?TLS_1_0) ->
```

File: lib/ssl/test/ssl_renegotiate_SUITE.erl [hunk 5]
```diff
@@ -536,9 +536,9 @@ renegotiate_rejected(Socket) ->
...
     (N div 2) + 1;
-treashold(N, _) ->
+treshold(N, _) ->
```
Comment (Maria-12648430) [inline lib/ssl/test/ssl_renegotiate_SUITE.erl:541]:
```suggestion
threshold(N, _) ->
```

**Review by Maria-12648430 — CHANGES_REQUESTED** [review]

**Comment by u3s** [comment]
Thanks @Maria-12648430 :-)

**Comment by u3s** [comment]
> Does this fix all warnings? If so why not turn on warnings as errors in the makefile so we don't introduce new ones.

Now we collectively recalled the ct auto_compile feature issue which kind of ignores the compilation result of a test suite and tries to execute no matter how bad the compilation went.

If the related beam file is found (could be result of previous succesful compilation) execution attempt will happen.

Therefore, I would like to merge this PR as is - just fixing the warnings.
And adding "warnings as errors" for test suites in the other internal issue for auto_compile feature. I will try to enable it for ssl and ssh.

**Pull request merged**